classes/inbox/Inbox.php

```diff
@@ -140,27 +137,31 @@ public static function getUserData(System $system, string $user_name): ?InboxUse
             user_name: $data['user_name'],
             avatar_link: $data['avatar_link'],
             forbidden_seal: $data['forbidden_seal'] ? json_decode($data['forbidden_seal'], true) : null,
-            staff_level: $data['staff_level'],
-            blocked_ids: json_decode($data['blocked_ids'], true)
+            staff_level: $data['staff_level']
         );
     }
 
     /**
      * @param InboxUser[] $convo_members
      * @return boolean
      */
-    public static function checkBlacklist(array $convo_members): bool {
+    public static function checkBlacklist(System $system, array $convo_members): bool {
         $user_ids = [];
         foreach($convo_members as $member) {
             $user_ids[$member->user_id] = $member->user_id;
         }
         foreach($convo_members as $member) {
-            if (empty($member->blocked_ids)) {
+            $blacklist = new Blacklist(
+                system: $system,
+                user_id: $member->user_id
+            );
```

Comment:
To consider - This will trigger an individual database query for every member of the conversation. How many times might `checkBlacklist` be called on one page load or API request?

Reply:
Disabling for now, I want to rethink my approach.